### PR TITLE
[Refactor] Release Version 3.0

### PR DESCRIPTION
**Pull request by g-cassie**
Changes are detailed in the new `readme.md`.  I have been working with @Emerson throughout on this so we are both on the same page. 

Some key points:

- You must be on Ember 2.0+  (1.13 probably works but we will not officially support it)

- Even better if you are on Ember 2.3+ so you can use the `hash` helper

- This is a major release so your app will break when you upgrade. The main thing you will need to do is change all your `type='input | select | textarea'` attrs to `widget='input | select | textarea'`.  

- Yes, we are using two-way bindings for form inputs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,35 +33,80 @@ You can see the basic dummy app here:
...
+you would expect and it is easy to add more custom components as
+needed.
+
+FM2000's primary entry point is the `fm-field` component.  This is
+used to render a form field to the page.  Even when you are making custom
+components you will still use `fm-field` to render them.
+
+Every time a field is rendered, it renders a `display` and a `widget`.
+`displays` are responible for the layout and css formatting of your form
+field.  For example, a using different `display`s you could opt to
```
Comment (Emerson) [inline README.md]: Small typo "a using"

File: README.md [hunk 2]
```diff
@@ -33,35 +33,80 @@ You can see the basic dummy app here:
...
+that is manipulated.  There are built in widgets for things like
+`input` and `textarea` but you could add a custom date picker widget or
+a fancy autocomplete widget.  To learn more about creating custom `displays` 
+and `widgets` see the **Extending Form Master** below.
+
+#### A Brief Not on What FM2000 is Not
+
+FM2000 is all about rendering interactive forms in a standardized
+way with a concise and extensible syntax.  FM2000 is *not* about doing
+form validation or managing CRUD logic.
```
Comment (Emerson) [inline README.md]: Not sure where we ended up with validation hooks, but could be worth mentioning here

File: README.md [hunk 3]
```diff
@@ -33,35 +33,80 @@ You can see the basic dummy app here:
...
+
+### Built-In Components
+
+The following is the basic API for Ember 2.3+.  The `hash` helper that
+is used below is not present in Ember 2.0-2.2. We are currently still supporting
+setting the widgetAttrs values for the default widgets directly on the
+`fm-field` component.  You can look at aso for those versions just
```
Comment (Emerson) [inline README.md]: Typo "aso"

File: README.md [hunk 4]
```diff
@@ -78,9 +123,135 @@ You can see the basic dummy app here:
...
+receive the following attrs:
+
+- `value`: the current value of the input.
+
+- `action` (optional): action to call with a new value chosen through
+  user interaction with the widget. If you do not want to support
+  actions-up/data-down design you can modify `value` directly and ignore
+  `action`.
+
+- `widgetAttrs`: an Ember.Object which is used as a vessel to pass any
```
Comment (Emerson) [inline README.md]: Love this clarity and use of the word vessel

File: README.md [hunk 5]
```diff
@@ -78,9 +123,135 @@ You can see the basic dummy app here:
...
+
+  {{#if helptext}}
+    <span class="{{styles.helptextClass}}">
+      {{helptext}}
+    </span>
+  {{/if}}
+</div>
+```
+
+This widget will be included wherever you place the `yield` statement.
```
Comment (Emerson) [inline README.md]: So good...

File: README.md [hunk 6]
```diff
@@ -78,9 +123,135 @@ You can see the basic dummy app here:
...
+```
+
+Note how we defer to `styles` property for class names where possible.
+This ties in the `fmConfig` service which provides easy configuration of
+form classes on a global basis.
+
+The best place to start for creating a display is to look at
+`fm-displays/default.hbs`.
+
 ### Errors
```
Comment (Emerson) [inline README.md]: Would love to add a couple notes on how people should approach client side vs server side, even if it's beyond the scope of this library

File: addon/components/fm-field.js [hunk 7]
```diff
@@ -1,82 +1,122 @@
...
+   *
+   * With Ember 2.3 adoption this should be phased out so that users
+   * pass `widgetAttrs` in directly using the `hash` helper.
+   *
+   * @method init
+   * @return {void}
+   **/
+  init(){
+    this._super();
+    WIDGET_ATTR_ALIASES.forEach(field => {
```
Comment (Emerson) [inline addon/components/fm-field.js]: Great!

**Comment by Emerson** [comment]
This is some really amazing work and represents a much clearer vision than what I had originally written.

If you're interested, I think we should consider moving this into its own Github org. Also going to chat with one of the designers at TWG about maybe getting a hot logo.

**Comment by g-cassie** [comment]
@Emerson Awesome thanks for the review. I am going to fix the typos. For now I will just remove everything talking about validation.  I have a vision for that too now - another big project but mainly drawing on code I have already written in my private repos.  I think we can make this a one stop form solution - but keep it pluggable so people can just use the UI stuff if they want or just use the validations stuff.

I don't have a strong opinion but maybe we could shorten the name to "Form Master"? Just because we are using the `fm` prefix everywhere so it effectively is the name that the addon is being referred to by.  This is not a strongly held opinion so feel free to say no to that.

**Comment by Emerson** [comment]
I think it's still worth leaving in the quip about validation and how passing errors works - just wasn't sure if you had any examples for client side validation a onUserInteraction hook or something. 

I'll think over the name stuff and we can connect soon and chat.

**Comment by g-cassie** [comment]
@Emerson great, just pushed an updated Readme with your changes and some other fixes.

**Pull request merged**